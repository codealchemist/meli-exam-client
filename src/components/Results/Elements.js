import styled from 'styled-components'
import Colors from '../Colors'

export const Wrapper = styled.div`
  display: flex;
  position: relative;
  border-bottom: 1px solid #eee;
  padding: 10px 0;
  cursor: pointer;
  animation: all 2x;

  :hover {
    border-bottom: 1px solid ${Colors.brand.main};
  }

  label {
    position: absolute;
    top: 40px;
    right: 40px;
  }
`

export const LeftWrapper = styled.div`
  width: 160px;
  margin: auto;
  text-align: center;
  padding: 0 10px;

  img {
    border-radius: 4px;
  }
`

export const RightWrapper = styled.div`
  width: 100%;

  h3 {
    font-size: 24px;
    margin-bottom: 0;
  }

  p {
    font-size: 20px;
  }
`
