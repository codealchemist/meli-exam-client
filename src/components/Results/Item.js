import React from 'react'
import { withRouter } from 'react-router-dom'
import { getProductDetails } from '../../api'
import Context from '../../Context'
import { Wrapper, LeftWrapper, RightWrapper, PriceRow } from './Elements'

const openDetails = ({ item, history, context }) => {
  const { id } = item
  console.log('Open Details', item)

  getProductDetails(id).then(response => {
    console.log('GOT product details response', response)
    context.setProductDetails(response)

    // Redirect to Details Page.
    history.push(`/items/${id}`)
  })
}

const Item = ({ item, history }) => (
  <Context.Consumer>
    {context => (
      <Wrapper onClick={() => openDetails({ item, history, context })}>
        <LeftWrapper>
          <img src={item.picture} alt="Product" />
        </LeftWrapper>

        <RightWrapper>
          <PriceRow>
            <h3>$ {item.price.amount}</h3>
            <label>{item.address.state_name}</label>
          </PriceRow>
          <p>{item.title}</p>
        </RightWrapper>
      </Wrapper>
    )}
  </Context.Consumer>
)

export default withRouter(Item)
