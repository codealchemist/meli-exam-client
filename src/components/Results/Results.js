import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Item from './Item'
import Colors from '../Colors'

const Wrapper = styled.div`
  max-width: 1200px;
  width: 100%;
  margin: auto;
  background: ${Colors.default.light1};
  color: ${Colors.default.dark1};
`

const Results = ({ items }) => (
  <Wrapper>
    {items.map(item => (
      <Item key={item.id} item={item} />
    ))}
  </Wrapper>
)

Results.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({}))
}

Results.defultProps = {
  items: []
}

export default Results
