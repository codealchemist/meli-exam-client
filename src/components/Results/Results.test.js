import React from 'react'
import { mount } from 'enzyme'
import Results from './Results'

jest.mock('./Item', () => 'div')

describe('Results', () => {
  it('should render without items', () => {
    const items = []
    const wrapper = mount(<Results items={items} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with items', () => {
    const items = [{ id: 42, name: 'rock' }]
    const wrapper = mount(<Results items={items} />)
    expect(wrapper).toMatchSnapshot()
  })
})
