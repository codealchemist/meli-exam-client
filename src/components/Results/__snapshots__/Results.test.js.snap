// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render with items 1`] = `
<Results
  items={
    Array [
      Object {
        "id": 42,
        "name": "rock",
      },
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "jvYsJV",
            "rules": Array [
              "
  max-width: 1200px;
  width: 100%;
  margin: auto;
  background: ",
              "#fff",
              ";
  color: ",
              "#333",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa jvYsJV"
      >
        <div
          item={
            Object {
              "id": 42,
              "name": "rock",
            }
          }
          key="42"
        />
      </div>
    </StyledComponent>
  </styled.div>
</Results>
`;

exports[`Results should render without items 1`] = `
<Results
  items={Array []}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "jvYsJV",
            "rules": Array [
              "
  max-width: 1200px;
  width: 100%;
  margin: auto;
  background: ",
              "#fff",
              ";
  color: ",
              "#333",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa jvYsJV"
      />
    </StyledComponent>
  </styled.div>
</Results>
`;
